src/Storages/MergeTree/MergeTreeBlockReadUtils.cpp

```diff
@@ -267,12 +266,14 @@ MergeTreeReadTaskColumns getReadTaskColumns(
     /// inject columns required for defaults evaluation
     bool should_reorder = !injectRequiredColumns(storage, metadata_snapshot, data_part, column_names).empty();
 
-    if (prewhere_info)
+    if (prewhere_info_list)
     {
-        if (prewhere_info->alias_actions)
-            pre_column_names = prewhere_info->alias_actions->getRequiredColumns();
-        else
-            pre_column_names = prewhere_info->prewhere_actions->getRequiredColumns();
+        for (const auto & prewhere_info : *prewhere_info_list)
```

Comment:
I think it is enough to get required columns only from first actions cause they are in chain.
Maybe it is not true, but I would rather check.
Also, it may probably fail if prewhere reuse row level filter result ...